### DISCUSSION TITLE
Suggentions for a Open Source lightweight db using with Logtalk?

### DISCUSSION THREAD

**Discussion by hnbeck** [LogtalkDotOrg/logtalk3#126] [Q&A]
Hi, 

one part of my project I'm working on is persistence of (3D) worlds. The basic file format is intended to be Universal Scene Description  (USD) from Pixar for serveral reasons. In consequence, this means to make 3D based objects and their properties persistent in a data base local to the Logtalk App.

Has anyone a suggestion which lightweight (Graph?, Key Value?) OpenSource DB works well with Logtalk? Lightweight is required because the per PC deployment of the app  should be a bundle of a game engine, the Logtalk & App plus the database in the end. The download and installation should be as easy as possible for the user. I also thought about of using the RDF persistence of SWI Prolog, but maybe even a simple extra DB (Maria, CouchDB ..... Neo4j....) is better, I don't know. 

If anyone has experience with Logtalk and DBs I really would appreciate to get something shared :)

Cheers and thx in advance

Hans

**Comment by pmoura** [comment]
Using a DB from Logtalk/Prolog in an efficient way usually requires using the Prolog native/proprietary foreign-language interface (FLI). That limits the choices.

**Comment by Stranger2015** [comment]
Hi, 

Have a look at
http://www.alachisoft.com/tayzgrid/
https://github.com/Alachisoft/TayzGrid
